### PR TITLE
feat(signalfx): add optional default scope and location keys when configuring accounts

### PR DESCRIPTION
**Pull request by fieldju**
This PR makes including `_scope_key` in the extendedScopeParameters optional and allows operators to define account / org wide defaults.

This PR also adds the location to the query if it is supplied as an account default or in the extendedScopeParameters as `_location_key`.

Both changes are backwards compatible.

### CODE REVIEW

File: kayenta-signalfx/src/test/java/com/netflix/kayenta/signalfx/metrics/SimpleSignalFlowProgramBuilderTest.java [hunk 1]
```diff
@@ -58,13 +60,88 @@ public void test_that_the_program_builder_builds_the_expected_program() {
     assertEquals(expected, builder.build());
   }
 
+  @Test
+  public void test_that_the_program_builder_builds_the_expected_program_when_location_is_provided_via_default() {
```
Comment (duftler) [inline kayenta-signalfx/src/test/java/com/netflix/kayenta/signalfx/metrics/SimpleSignalFlowProgramBuilderTest.java]: This test and the one just below seem to match except for where the location key is specified. Could combine them and use a `where:` block and `@Unroll` instead. Would also make it easier for a reader to tease out the differences.

File: kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/config/SignalFxScopeConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class SignalFxScopeConfiguration {
+
+  private String defaultScopeKey;
+  private String defaultLocationKey;
+
```
Comment (duftler) [inline kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/config/SignalFxScopeConfiguration.java]: Extraneous blank line.

File: kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/metrics/SignalFxMetricsService.java [hunk 3]
```diff
@@ -80,13 +84,15 @@ public String buildQuery(String metricsAccountName,
                            CanaryMetricConfig canaryMetricConfig,
                            CanaryScope canaryScope) {
 
-    SignalFxCanaryScope signalFxCanaryScope = (SignalFxCanaryScope)canaryScope;
+    SignalFxScopeConfiguration scopeConfiguration = signalFxScopeConfigurationMap.get(metricsAccountName);
+
```
Comment (duftler) [inline kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/metrics/SignalFxMetricsService.java]: Extraneous blank line.

File: kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/config/SignalFxScopeConfiguration.java (hunk 2 above)
Comment (fieldju) [reply kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/config/SignalFxScopeConfiguration.java]: - [X] done.

File: kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/metrics/SignalFxMetricsService.java (hunk 3 above)
Comment (fieldju) [reply kayenta-signalfx/src/main/java/com/netflix/kayenta/signalfx/metrics/SignalFxMetricsService.java]: - [X] done.

**Pull request merged**